src/l500/l500-options.cpp

```diff
@@ -64,27 +94,63 @@ namespace librealsense
         auto min_value = float(*(reinterpret_cast<int32_t*>(min.data())));
 
         _range = option_range{ min_value,
-            max_value,
-            float(*(reinterpret_cast<int32_t*>(step.data()))),
-            def };
+                               max_value,
+                               float( *( reinterpret_cast< int32_t * >( step.data() ) ) ),
+                               query_default() };
+    }
+
+    void l500_hw_options::update_default( float def )
+    {
+        _range.def = def;
     }
 
     void l500_hw_options::enable_recording(std::function<void(const option&)> recording_action)
     {}
 
-    float l500_hw_options::query(int mode) const
+    float l500_hw_options::query( l500_command op, int mode ) const
```

Comment:
Maybe separate into `query_current()` and `query_default()`? We don't really need to have one that accepts an argument, do we?